pkg/karmadactl/cmdinit/config/config_test.go

```diff
@@ -0,0 +1,305 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const testConfig = `
+apiVersion: config.karmada.io/v1alpha1
+kind: KarmadaInitConfig
+metadata:
+ name: karmada-init
+spec:
+ certificates:
+   caCertFile: "/etc/karmada/pki/ca.crt"
+   caKeyFile: "/etc/karmada/pki/ca.key"
+   externalDNS:
+     - "localhost"
+     - "example.com"
+   externalIP:
+     - "192.168.1.2"
+     - "172.16.1.2"
+   validityPeriod: "8760h0m0s"
+ etcd:
+   local:
+     imageRepository: "karmada/etcd"
```

Comment:
`karmada/etcd` does not seem to be an efficient container repository

Reply:
The repository field here is an error, my test case documentation here is automatically generated, there are some errors, I will correct

Reply:
The problem here has been modified, and I hope to get your further guidance

Reply:
The problem here has been modified, and I hope to get your further guidance
@liangyuanpeng